### ISSUE TITLE
Unit test invoking a failing `Debug.Assert` indirectly runs indefinitely in VS unit test explorer

### ISSUE THREAD

**Issue by Rekkonnect** [thomhurst/TUnit#596]
Assume the following test code:

```csharp
public sealed class BugTest
{
    [Test]
    public async Task SomeTest()
    {
        FailingMethod();
    } 

    private void FailingMethod()
    {
        Debug.Assert(true is false);
    }
}
```

This test will run indefinitely, since the assertion fails. A successful `Debug.Assert` will not hang the test run.

This was observed in the unit test runner in Visual Studio Preview. Using `dotnet test` will produce logs showing that the process was terminated due to the assertion failure. The CLI seems to be resilient to process terminations while the unit test explorer expects that the process will remain alive.

**Comment by thomhurst** [comment]
@Rekkonnect I don't believe I can do anything to fix this? That Debug.Assert call will terminate the process if it fails: https://learn.microsoft.com/en-us/dotnet/api/system.environment.failfast?view=net-8.0

If the process has terminated, then I simply can't do anything.

I guess if anything this should be a VS bug so that if the dotnet process has died, they should stop showing the test as running.

**Comment by Rekkonnect** [comment]
You're right about this being a VS bug, however I believe the process terminating is a resiliency issue in the framework. 

Reading through the docs for Debug.Assert it seems that it can be caught:
https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.debug.assert?view=net-8.0

If that's the case, I believe TUnit should handle it and avoid killing the process

**Comment by thomhurst** [comment]
Yep you're right @Rekkonnect - Thanks for the insight. I've managed to hook into it. This should be fixed in 0.1.801
